### PR TITLE
Fix bugs for uploading

### PR DESCRIPTION
**Pull request by shanitang**
Fix bugs for uploading with `filename` and `body`

@zendesk/archer

### CODE REVIEW

**Comment by kssmith** [comment]
Any update on this getting merged? We're running into problems on our production.

File: src/Zendesk/API/Http.php [hunk 1]
```diff
@@ -87,8 +87,11 @@ public static function send(Client $client, $endPoint, $json = array(), $method
                 $size = filesize($filename);
                 $fileData = fread($file, $size);
                 $json = $fileData;
-                $contentType = '';
```
Comment (mikealmond) [inline src/Zendesk/API/Http.php]: Need to tell cURL the file size as well

```php
curl_setopt($curl, CURLOPT_INFILESIZE, $size);
```

File: src/Zendesk/API/Http.php [hunk 2]
```diff
@@ -87,8 +87,11 @@ public static function send(Client $client, $endPoint, $json = array(), $method
...
-            }            
+            }
+            else if (isset($json['body'])){
+                $json = $json['body'];
```
Comment (mikealmond) [inline src/Zendesk/API/Http.php]: Need to set the size here

```php
curl_setopt($curl, CURLOPT_INFILESIZE, strlen($json['body']));
```

File: src/Zendesk/API/Http.php [hunk 3]
```diff
@@ -87,8 +87,11 @@ public static function send(Client $client, $endPoint, $json = array(), $method
...
+            }
+
             $httpHeader[] = 'Content-Type: '.$contentType;
         } else {
             $contentType = '';
```
Comment (mikealmond) [inline src/Zendesk/API/Http.php]: Does `$json` need to be reset with `$json['uploaded_data']`? It was before the change to this file
https://github.com/zendesk/zendesk_api_client_php/blob/552a5cad477b53f7ac7217c41c94529da8f28f2b/src/Zendesk/API/Http.php

File: src/Zendesk/API/Http.php (hunk 1 above)
Comment (shanitang) [inline src/Zendesk/API/Http.php]: @mikealmond 
Yes, you're right. Thanks!
But do you know why it still works without those set up in this case?

**Pull request merged**